checker.c

```diff
@@ -1,20 +1,34 @@
 #include <stdio.h>
 #include <assert.h>
-
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+ 
```

Comment:
if/else range check is repeated in all the parameter check function. It appears that DRY principle violated